plugins/tflint/plugin.py

```diff
@@ -0,0 +1,195 @@
+            if recursive == "true":
+                options.append("--recursive")
+
+            return_code, stdout, stderr = self.run_cli("tflint", "--init")
+            if return_code != 0:
+                self.logger.error(f"tflint --init failed with code {return_code}")
+                if stderr:
+                    # Display stderr manually because output display is disabled
+                    self.logger.error("\n".join(stderr))
+                exit(1)
+
+            return_code, stdout, stderr = self.run_cli(
+                "tflint", *options, print_output=False
+            )
+            stdout_json = json.loads("\n".join(stdout))
+
+            # KLUDGE: The `expect_code` argument does not support multiple values so we need to manually handle this.
+            # `0` means success and `2` means issues found which is informational, not a failure.
+            if return_code not in [0, 2]:
+                for error in stdout_json["errors"]:
+                    self.logger.error(error["message"])
+                exit(1)
+
+            self.add_to_policy_input("tflint", stdout_json)
+
+            if len(stdout_json["issues"]) == 0:
+                self.logger.info("No issues found")
+                return
+
+            findings = {
+                "error": {},
+                "warning": {},
+                "notice": {},
+            }
+            for match in stdout_json["issues"]:
+                severity = match["rule"]["severity"]
+                rule_name = match["rule"]["name"]
+
+                if rule_name not in findings[severity]:
+                    findings[severity][rule_name] = []
+
+                findings[severity][rule_name].append(match)
+
+            markdown = "# TFLint Findings\n\n"
+            for severity, rules in findings.items():
+                # Skip severity level if no issues were found
+                if len(rules) == 0:
+                    continue
+
+                emoji = None
+                if severity == "notice":
+                    emoji = "🟡"
+                elif severity == "warning":
+                    emoji = "🟠"
+                elif severity == "error":
+                    emoji = "🔴"
+                if emoji is not None:
+                    markdown += f"## {emoji} {severity.title()} Findings\n\n"
+                else:
+                    markdown += f"## {severity.title()} Findings\n\n"
+
+                for rule_name, issues in rules.items():
+                    markdown += f"### {rule_name}\n\n"
+                    for issue in issues:
+                        markdown += f"- {issue['message']} _({issue['range']['filename']}:{issue['range']['start']['line']})_\n"
+                    markdown += "\n"
+
+            result = self.send_markdown(markdown)
+            if result:
+                self.logger.info(
+                    "Issues found. Check the Plugins Output tab for details."
```

Comment:
If issues are found should we return non zero?

Reply:
I wondered too and decided to use the policy because failing the run from a hook makes the reason less obvious (some people ignore the logs) while a failure triggered by a policy is very obvious in the UI.